source/module_hsolver/diagh.h

```diff
@@ -0,0 +1,31 @@
+#ifndef DIAGH_H
```

Comment:
One small question: when shall we use `diago` (e.g. `diago_cg`), and when shall we use `diag` (e.g. `diagh` here)?

Reply:
Good question, I don't know either. In old code, we use both "diag" and "diago" for abbreviation of diagonalization, what is your suggestion?

Reply:
> Good question, I don't know either. In old code, we use both "diag" and "diago" for abbreviation of diagonalization, what is your suggestion?

In math formula, "diag(M)" is used to denote the [diagonal matrix](https://en.wikipedia.org/wiki/Diagonal_matrix). So  I think following this convention is better, and totally get rid of "diago".

Reply:
At this time, I think it is ok to keep same format as old file diago_cg and diago_david, we can discuss it in the future.